flybywiresim/a32nx
# fix(pfd): Fix PFD altitude indications

Fixes #[issue_no]

## Summary of Changes
- Fixes current metric altitude being shown in cyan, rather than green.
- Fixes selected metric altitude showing in green rather than cyan, magenta or white.
- Fixes FCU altitude being constrained after GS capture.

Something I noticed that is not fixed by this PR is that the selected altitude indications on the PFD (both ft and meters) might briefly flicker to the FCU altitude before correctly showing a constraint altitude in magenta. I don't think this is supposed to happen.

## Screenshots (if necessary)


Before:

![](https://media.discordapp.net/attachments/754851312306487326/1074768545176633424/20230213_200217.jpg?width=1026&height=702)
![before-fix-no-mda](https://user-images.githubusercontent.com/22713769/219768814-573c113f-228d-45e6-9b93-a9c8d38008f7.png)

After:
![after-fix-no-mda](https://user-images.githubusercontent.com/22713769/219768915-643b4ef4-2a96-4c36-b523-c86fcc571d1d.png)
![after-fix-on-gs](https://user-images.githubusercontent.com/22713769/219768900-a8a03227-76cb-4990-a48a-f835a3106ade.png)
![after-fix-below-mda](https://user-images.githubusercontent.com/22713769/219768909-43c927f0-3450-4954-ae82-9e9c88c8f851.png)

## References

- https://youtu.be/HaizWlqYGr8
- https://youtu.be/JaTyrHFSXa4

Reference for selected metric altitude indicator in magenta.
![image](https://user-images.githubusercontent.com/22713769/219714770-9f559304-360a-413c-adcb-45ef0bb95ce7.png)

Reference for selected metric indicator in cyan
![image](https://user-images.githubusercontent.com/22713769/219714889-ac5da72b-2b5d-45a2-b45f-813c102513d6.png)

Reference for current metric altitude below minimums
![yt-pfd-metric-alt-below-minimum](https://user-images.githubusercontent.com/22713769/219715352-acd18a58-85d4-4b0d-afe1-19266aaf471b.png)




## Additional context



Discord username (if different from GitHub): BlueberryKing#6641

## Testing instructions

- Load up flight, ensure metric altitude indication is shown in green.
- Make sure metric altitude indication turns amber once below MDA
- Make sure selected metric altitude indication is cyan, magenta or white, matching the color of the selected metric altitude.
- Fly an ILS approach. Once the GS is captured. Make sure that the selected altitude indication (both metric and ft) matches the FCU value and is shown in white.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
